torch/_inductor/util_autotuning_log_parser.py

```diff
@@ -0,0 +1,244 @@
+        match: Optional[re.Match[str]] = re.search(
+            r"Layout\('cuda', torch\.([^, ]+),", layout_str
+        )
+        assert match is not None, f"Failed to parse dtype: {layout_str}"
+        return match.group(1)
+
+    @classmethod
+    def parse_record(cls, rec: Dict[str, Any]):
+        result = dict()
+        result["backend"] = rec["backend"]
+        result["name"] = rec.get("name", "ATen")
+        inputs = json.dumps(rec["input_nodes"])
+        result["problem_hash"] = cls.strhash(inputs)
+        result["kernel_schedule"] = rec.get("kernel_schedule", "")
+        result["tile_shape"] = rec.get("tile_shape", "[]")
+        result["benchmark_result"] = rec["benchmark_result"]
+        result["device"] = rec.get("device", "unknown")
+        result["cuda_device_name"] = rec["cuda_device_name"]
+        # result['keys'] = ",".join(rec.keys())
+        for name, tinfo in cls.parse_inputs(rec["input_nodes"], rec["backend"]).items():
+            if isinstance(tinfo, dict):
+                for key, value in tinfo.items():
+                    result[f"{name}_{key}"] = value
+            elif tinfo is not None:
+                result[name] = tinfo
+        return result
+
+    @staticmethod
+    def load_raw_records(path_or_fd: Union[str, io.TextIOWrapper]):
+        if isinstance(path_or_fd, str):
+            with open(path_or_fd, encoding="utf-8") as fh:
+                for line in fh:
+                    yield json.loads(line)
+        else:
+            if hasattr(path_or_fd, "seek"):
+                path_or_fd.seek(0)
+            for line in path_or_fd:
+                yield json.loads(line)
+
+    @classmethod
+    def load_records(
+        cls,
+        path_or_fd: Union[str, io.TextIOWrapper, List[Union[str, io.TextIOWrapper]]],
+    ):
+        if isinstance(path_or_fd, list):
+            for pf in path_or_fd:
+                yield from cls.load_records(pf)
+        else:
+            for record in cls.load_raw_records(path_or_fd):
+                yield cls.parse_record(record)
+
+    @staticmethod
+    def load_raw_grepped_records(path, pattern):
+        with open(path, encoding="utf-8") as fh:
+            for line in fh:
+                if pattern in line:
+                    yield json.loads(line)
+
+    @classmethod
+    def parse_layout(cls, layout_str):
+        match: Optional[re.Match[str]] = re.search(
+            r"size=\[([0-9]+), ([0-9]+)(, ([0-9]+))?\], stride=\[([0-9]+), ([0-9]+)(, ([0-9]+))?\]",
+            layout_str,
+        )
+        assert match is not None, f"Failed to parse layout: {layout_str}"
+        vals = match.groups()
+        res: Dict[str, Any] = {}
+        if vals[2] is None:
+            assert vals[6] is None
+            res = {
+                "size": (1, int(vals[0]), int(vals[1])),
+                "stride": (0, int(vals[4]), int(vals[5])),
+            }
+        else:
+            res = {
+                "size": (int(vals[0]), int(vals[1]), int(vals[3])),
```

Comment:
Won't the casts to ints here fail with dynamic shapes?

Reply:
Yes. Using the parser on dynamic shapes would be pointless, though. The point of this utility class (as I see it) is to enable microbenchmark analysis where we have obtained timings for specific fixed shapes. While this could be made to gracefully reject entries with dynamic shapes, or we could try to convert all dynamic shapes to specific shapes by using shape hints, I don't think it's worth to invest the time to develop and test that change. The primary usecase for this is when doing experiments with dynamic_shapes=False. I will add a comment that clarifies this. Please tell me if you see a usecase for dynamic_shapes=True, though. We could also add dynamic shape support in a separate PR if we see a need.